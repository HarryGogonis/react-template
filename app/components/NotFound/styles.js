export default {
  width: '420px',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  header: {
    borderBottom: '1px solid #ccc',
    paddingBottom: '25px',
  },
};
