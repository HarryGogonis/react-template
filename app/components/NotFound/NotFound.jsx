import React from 'react';
import BasicLayout from 'components/BasicLayout';
import styles from './styles';

const NotFound = () =>
  <BasicLayout>
    <div style={styles}>
      <h2 style={styles.header}>404 Page - Not Found</h2>
      <p>The page you requested was not found.</p>
    </div>
  </BasicLayout>;

export default NotFound;
