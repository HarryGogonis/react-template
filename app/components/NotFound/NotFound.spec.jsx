import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NotFound from './NotFound';

jest.mock('components/BasicLayout', () => 'BasicLayout');

let wrapper;

describe('<NotFound />', () => {
  beforeEach(() => {
    wrapper = shallow(<NotFound />);
  });

  it('is using the BasicLayout', () => {
    expect(wrapper.type()).toBe('BasicLayout');
  });

  it('a single child element is passed to the layout', () => {
    expect(wrapper.children.length).toBe(1);
  });

  it('a header and paragraph', () => {
    expect(wrapper.find('h2').length).toBe(1);
    expect(wrapper.find('p').length).toBe(1);
  });

  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<NotFound />);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
