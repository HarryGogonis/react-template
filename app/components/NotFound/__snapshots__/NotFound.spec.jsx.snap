// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should match the snapshot 1`] = `
<BasicLayout>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "header": Object {
          "borderBottom": "1px solid #ccc",
          "paddingBottom": "25px",
        },
        "width": "420px",
      }
    }
  >
    <h2
      style={
        Object {
          "borderBottom": "1px solid #ccc",
          "paddingBottom": "25px",
        }
      }
    >
      404 Page - Not Found
    </h2>
    <p>
      The page you requested was not found.
    </p>
  </div>
</BasicLayout>
`;
